net/rpc/server.go

```diff
@@ -191,13 +191,34 @@ type Server struct {
 	freeReq    *Request
 	respLock   sync.Mutex // protects freeResp
 	freeResp   *Response
+
+	serverServiceCallInterceptor ServerServiceCallInterceptor
 }
 
 // NewServer returns a new Server.
 func NewServer() *Server {
-	return &Server{}
+	return NewServerWithOpts()
+}
+
+// NewServerWithOpts returns a new Server with the following functional options.
+func NewServerWithOpts(options ...func(*Server)) *Server {
+	s := &Server{}
+	for _, option := range options {
+		option(s)
+	}
+	return s
+}
+
+func WithServerServiceCallInterceptor(interceptor ServerServiceCallInterceptor) func(*Server) {
+	return func(s *Server) {
+		s.serverServiceCallInterceptor = interceptor
+	}
```

Comment:
`NewServerWithOpts` take multiple options but we only support 1 interceptor per server, what is the reasoning here?

Reply:
See above/ here: https://github.com/hashicorp/consul-net-rpc/pull/8#discussion_r814872224